src/ruler.js

```diff
@@ -91,12 +109,19 @@ export class DragRulerRuler extends Ruler {
 		this.labels.removeChildren().forEach(c => c.destroy());
 	}
 
-	dragRulerDeleteWaypoint(event={preventDefault: () => {return}}) {
+	dragRulerDeleteWaypoint(event={preventDefault: () => {return}}, options={snap: true}) {
 		if (this.waypoints.filter(w => !w.isPrevious).length > 1) {
 			event.preventDefault();
 			const mousePosition = canvas.app.renderer.plugins.interaction.mouse.getLocalPosition(canvas.tokens);
 			const rulerOffset = this.rulerOffset;
-			this._removeWaypoint({x: mousePosition.x + rulerOffset.x, y: mousePosition.y + rulerOffset.y});
+
+			if(options.snap === undefined){
```

Comment:
If we already set `snap: true` as default further above, this code shouldn't be necessary.